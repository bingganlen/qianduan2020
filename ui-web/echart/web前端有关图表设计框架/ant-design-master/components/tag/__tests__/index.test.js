import React from 'react';
import { mount } from 'enzyme';
import Tag from '..';

describe('Tag', () => {
  beforeAll(() => {
    jest.useFakeTimers();
  });

  afterAll(() => {
    jest.useRealTimers();
  });

  it('should be closable', () => {
    const onClose = jest.fn();
    const wrapper = mount(<Tag closable onClose={onClose} />);
    expect(wrapper.find('.anticon-close').length).toBe(1);
    expect(wrapper.find('div.ant-tag:not(.ant-tag-hidden)').length).toBe(1);
    wrapper.find('.anticon-close').simulate('click');
    expect(onClose).toBeCalled();
    jest.runAllTimers();
    wrapper.update();
    expect(wrapper.find('div.ant-tag:not(.ant-tag-hidden)').length).toBe(0);
  });

  it('should not be closed when prevent default', () => {
    const onClose = e => {
      e.preventDefault();
    };
    const wrapper = mount(<Tag closable onClose={onClose} />);
    expect(wrapper.find('.anticon-close').length).toBe(1);
    expect(wrapper.find('div.ant-tag:not(.ant-tag-hidden)').length).toBe(1);
    wrapper.find('.anticon-close').simulate('click');
    jest.runAllTimers();
    expect(wrapper.find('div.ant-tag:not(.ant-tag-hidden)').length).toBe(1);
  });

  it('should not call afterClose when change visible from false to true', () => {
    const handleAfterClose = jest.fn();
    const wrapper = mount(<Tag visible afterClose={handleAfterClose} />);
    wrapper.setProps({ visible: false });
    jest.runAllTimers();
    expect(handleAfterClose).toHaveBeenCalled();
    handleAfterClose.mockRestore();
    wrapper.setProps({ visible: true });
    jest.runAllTimers();
    expect(handleAfterClose).not.toHaveBeenCalled();
  });

  describe('visibility', () => {
    it('can be controlled by visible with visible as initial value', () => {
      const wrapper = mount(<Tag visible />);
      expect(wrapper.render()).toMatchSnapshot();
      wrapper.setProps({ visible: false });
      jest.runAllTimers();
      expect(wrapper.render()).toMatchSnapshot();
      wrapper.setProps({ visible: true });
      jest.runAllTimers();
      expect(wrapper.render()).toMatchSnapshot();
    });

    it('can be controlled by visible with hidden as initial value', () => {
      const wrapper = mount(<Tag visible={false} />);
      expect(wrapper.render()).toMatchSnapshot();
      wrapper.setProps({ visible: true });
      jest.runAllTimers();
      expect(wrapper.render()).toMatchSnapshot();
      wrapper.setProps({ visible: false });
      jest.runAllTimers();
      expect(wrapper.render()).toMatchSnapshot();
    });
  });
});
