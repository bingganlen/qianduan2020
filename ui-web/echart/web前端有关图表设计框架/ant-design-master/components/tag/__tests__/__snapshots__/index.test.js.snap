// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag visibility can be controlled by visible with hidden as initial value 1`] = `
<div
  class="ant-tag ant-tag-hidden"
/>
`;

exports[`Tag visibility can be controlled by visible with hidden as initial value 2`] = `
<div
  class="ant-tag"
/>
`;

exports[`Tag visibility can be controlled by visible with hidden as initial value 3`] = `
<div
  class="ant-tag ant-tag-hidden"
/>
`;

exports[`Tag visibility can be controlled by visible with visible as initial value 1`] = `
<div
  class="ant-tag"
/>
`;

exports[`Tag visibility can be controlled by visible with visible as initial value 2`] = `
<div
  class="ant-tag ant-tag-hidden"
/>
`;

exports[`Tag visibility can be controlled by visible with visible as initial value 3`] = `
<div
  class="ant-tag"
/>
`;
